File: SampleJavaTests/src/test/java/pageObjects/LoginPage.java
package pageObjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;


public class LoginPage extends PageObjectBase {


    public static WebElement username_txtbox(WebDriver driver){
        WebElement element = findElement(findByXpath("//input[contains(@ng-model,'login.username')]"), driver);
        return element;
    }

    public static WebElement password_txtbox(WebDriver driver){
        WebElement element = findElement(findByXpath("//input[contains(@ng-model,'login.password')]"), driver);
        return element;
    }

    public static WebElement login_btn(WebDriver driver){
        WebElement element = findElement(findById("start_shopping_login_button"), driver);
        return element;
    }

    public static WebElement signUp_btn(WebDriver driver){
        WebElement element = findElement(findByXpath("//button[contains(text(), 'Sign up')]"), driver);
        return element;
    }

    public static WebElement signUpName_txtbox(WebDriver driver){
        WebElement element = findElement(findByXpath("//input[contains(@ng-model,'register.name')]"), driver);
        return element;
    }



}

File: SampleTest/pages/loginPage.js
import BasePage from './basePage';

class LoginPage extends BasePage {
    constructor() {
        super();
        this.userId = element(by.model('login.username'));
        this.password = element(by.model('login.password'));
        this.loginButton = element(by.id('start_shopping_login_button'));
        this.signUpButton = element(by.xpath("//button[contains(text(), 'Sign up')]"));
        this.signUpNameTxtBox = element(by.model('register.name'));
        this.url = 'https://shop.shipt.com';
        this.pageLoaded = this.isVisible(this.loginButton);
    }

}
export default new LoginPage();